pkg/compute/rust.go

```diff
@@ -201,15 +216,21 @@ func (r Rust) Build(out io.Writer) error {
 	_, errStderr = io.Copy(stderr, stderrIn)
 	wg.Wait()
 
-	// Wait for the command to exit.
-	if err := cmd.Wait(); err != nil {
-		return fmt.Errorf("error during compilation process: %w", err)
-	}
 	if errStdout != nil {
-		return fmt.Errorf("error during compilation process: %w", errStdout)
+		return fmt.Errorf("error streaming stdout output from child process: %w", errStdout)
 	}
 	if errStderr != nil {
-		return fmt.Errorf("error during compilation process: %w", errStderr)
+		return fmt.Errorf("error streaming stderr output from child process: %w", errStderr)
+	}
+
+	// Wait for the command to exit.
+	if err := cmd.Wait(); err != nil {
+		// If we're not in verbose mode return the bufferred stderr output
+		// from cargo as the error.
+		if !verbose && stderrBuf.Len() > 0 {
+			return fmt.Errorf("error during compilation process:\n%s", strings.TrimSpace(stderrBuf.String()))
+		}
+		return fmt.Errorf("error during compilation process")
```

Comment:
This line swallows the `err` from `cmd.Wait()` but I think it's usually just "exited with status 1` so 🤷

Reply:
Yup this was intentional as the message wasn't helpful at all as you suggest. 👍